### PR TITLE
feat: Swap to Popover for Notebooks

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

The Sidebar concept was pretty limiting from a UX perspective. Thought I'd try the Popover style again and it feels much much nicer

## Changes

* Gets rid of the fixed sidebar in favour of a "popover"
* Nicer drag and drop feel as it "peeks" out when you start dragging
* Makes this work for all links

I think we might still have a fixed side area later for things like the "preview" notebooks but this Popover would fit much better for placeholder notebooks like the idea we have for the improved PersonsModal

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

**Comment by daibhin** [comment]
Really like this approach and feels like the right change given it moves us closer to a 3000 world.

Couple of things that we've chatted about a bit already:
- We lose side-by-side editing ability. This is minor compared to the impact of this change but something we should account for in peoples experience.
- When dragging a node to the peeking view you have no way of specifying which notebook to add the node to. Before you could choose / create a notebook when the sidebar was open. Feels like there should be separate drop areas at the bottom of the notebook for "add to existing" or "create new"

File: frontend/src/scenes/notebooks/Notebook/NotebookPopover.tsx [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+export function NotebookPopover(): JSX.Element {
+    const { visibility, shownAtLeastOnce, fullScreen, selectedNotebook, initialAutofocus, dropListeners } =
+        useValues(notebookPopoverLogic)
+    const { setVisibility, setFullScreen, selectNotebook, setElementRef } = useActions(notebookPopoverLogic)
+    const { createNotebook } = useActions(notebooksListLogic)
+    const { notebook } = useValues(notebookLogic({ shortId: selectedNotebook }))
+
+    const ref = useRef<HTMLDivElement>(null)
+
+    // NOTE: This doesn't work for some reason, possibly due to the way the editor is rendered
```
Comment (daibhin) [inline frontend/src/scenes/notebooks/Notebook/NotebookPopover.tsx:25]: I'm not sure I understand this comment. This seems to close fullscreen for me

**Review by daibhin — APPROVED** [review]
One tiny thing but this looks great!

File: frontend/src/scenes/notebooks/Notebook/NotebookPopover.tsx (hunk 1 above)
Comment (benjackwhite) [reply frontend/src/scenes/notebooks/Notebook/NotebookPopover.tsx]: Ah yeah I just implemented it wrong. Updated now to exit full screen and then hide it alltogether

**Pull request merged**